homeassistant/components/keyboard_remote.py

```diff
@@ -106,34 +82,60 @@ def _stop_keyboard_remote(_event):
 class KeyboardRemote(threading.Thread):
     """This interfaces with the inputdevice using evdev."""
 
-    def __init__(self, hass, device_descriptor, key_value):
+    def __init__(self, hass, config):
         """Construct a KeyboardRemote interface object."""
-        from evdev import InputDevice
+        from evdev import InputDevice, list_devices
 
-        self.device_descriptor = device_descriptor
-        try:
-            self.dev = InputDevice(device_descriptor)
-        except OSError:  # Keyboard not present
+        self.device_descriptor = config.get(DEVICE_DESCRIPTOR)
+        self.device_name = config.get(DEVICE_NAME)
+        self.dev = self._get_keyboard_device()
+        if self.dev is not None:
             _LOGGER.debug(
-                'KeyboardRemote: keyboard not connected, %s',
-                self.device_descriptor)
-            self.keyboard_connected = False
+                'KeyboardRemote: keyboard connected'
+                )
         else:
-            self.keyboard_connected = True
+            id_folder = '/dev/input/by-id/'
+            device_names = [InputDevice(file_name).name
+                            for file_name in list_devices()]
             _LOGGER.debug(
-                'KeyboardRemote: keyboard connected, %s',
-                self.dev)
+                'KeyboardRemote: keyboard not connected.\
+                Check /dev/input/event* permissions.\
+                Possible device names are:\n %s.\n \
+                Possible device descriptors are %s:\n %s',
+                device_names,
+                id_folder,
+                os.listdir(id_folder)
+                )
 
         threading.Thread.__init__(self)
         self.stopped = threading.Event()
         self.hass = hass
-        self.key_value = key_value
+        self.key_value = KEY_VALUE.get(config.get(TYPE, 'key_up'))
+
+    def _get_keyboard_device(self):
+        from evdev import InputDevice, list_devices
+        if self.device_name:
+            devices = [InputDevice(file_name) for file_name in list_devices()]
+            for device in devices:
+                if self.device_name == device.name:
+                    return device
```

Comment:
Mmh, now that I think of it,  actually this function shouldn't write to the log at all: it gets repeatedly called from the event loop, when the keyboard is not connected.